service/worker/deployment/deployment_client.go

```diff
@@ -369,51 +338,396 @@ func (d *DeploymentClientImpl) ListDeployments(ctx context.Context, namespaceEnt
 
 }
 
-// GenerateStartWorkflowPayload generates start workflow execution payload
-func (d *DeploymentClientImpl) generateStartWorkflowPayload(namespaceEntry *namespace.Namespace, deployment *deploymentpb.Deployment) (*commonpb.Payloads, error) {
-	workflowArgs := &deploymentspb.DeploymentWorkflowArgs{
+func (d *DeploymentClientImpl) SetCurrentDeployment(
+	ctx context.Context,
+	namespaceEntry *namespace.Namespace,
+	deployment *deploymentpb.Deployment,
+	updateMetadata *deploymentpb.UpdateDeploymentMetadata,
+	identity string,
+	requestID string,
+) (*deploymentpb.DeploymentInfo, *deploymentpb.DeploymentInfo, error) {
+	updatePayload, err := sdk.PreferProtoDataConverter.ToPayloads(&deploymentspb.SetCurrentDeploymentArgs{
+		Identity:       identity,
+		BuildId:        deployment.BuildId,
+		UpdateMetadata: updateMetadata,
+		RequestId:      requestID,
+	})
+	outcome, err := d.updateWithStartDeploymentSeries(
+		ctx,
+		namespaceEntry,
+		deployment.SeriesName,
+		&updatepb.Request{
+			Input: &updatepb.Input{Name: SetCurrentDeployment, Args: updatePayload},
+			Meta:  &updatepb.Meta{UpdateId: requestID, Identity: identity},
+		},
+		identity,
+		requestID,
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if failure := outcome.GetFailure(); failure.GetApplicationFailureInfo().GetType() == errNoChangeType {
+		return nil, nil, serviceerror.NewAlreadyExist(fmt.Sprintf("Build ID %q is already current for %q",
+			deployment.BuildId, deployment.SeriesName))
+	} else if failure != nil {
+		// TODO: is there an easy way to recover the original type here?
+		return nil, nil, serviceerror.NewInternal(failure.Message)
+	}
+
+	success := outcome.GetSuccess()
+	if success == nil {
+		return nil, nil, serviceerror.NewInternal("outcome missing success and failure")
```

Comment:
log?

Reply:
I'd rather not add just assorted logs everywhere. we should be more comprehensive: figure out which operations deserve logs, then find a few spots that can be sort of like interceptors for these clients (maybe a timer object + defer at the top of the call, similar to our metrics interceptors). it'd be good to get the metrics too.